Merged tool

jeremymeiss opened the pull request:
Merged the Subsampling and Multiprocessing Aspects into one Skript

Commit:
add subsampling

Commit:
add subsampling

jeremymeiss committed:
add subsampling

jeremymeiss committed:
"add subsampling"

jeremymeiss committed:
add subsampling

jeremymeiss committed:
update

jeremymeiss committed:
updated sample size, now working

jeremymeiss committed:
added comments to the skript, removed the debugg-
code. made an extra file for the sampler version

jeremymeiss committed:
Merged Multiprocessing Apsect with Subsampling
for a complete Tool

sjanssen2 commented on .DS_Store:
why do you want to check in these binary files?!

sjanssen2 commented on vxdetector/testlog:
why does this log file need to be part of the repo?

sjanssen2 commented on vxdetector/VXdetector.py:
```diff
@@ -4,246 +4,207 @@
+    print(f"BIOM file successfully converted to FASTQ format at: {fastq_output}")
+
+
+def sample_fastq(file_path, sample_size, sampled_indices=None):
+    '''Get random parts from the FASTQ file based on shared indices for paired-end reads.'''
+    sampled_reads = []  # List to hold sampled reads
+    open_func = gzip.open if file_path.endswith('.gz') else open
+
+    # Count total reads in the file by iterating line by line
+    with open_func(file_path, 'rt') as f:
+        total_reads = sum(1 for _ in f) // 4
+
+    # Adjust sample_size if it's greater than total_reads
+    if sample_size > total_reads:
+        sample_size = total_reads
+
+    if sampled_indices is None:
+        sampled_indices = sorted(random.sample(range(total_reads), sample_size))
+
+    with open_func(file_path, 'rt') as f:
+        read_idx = 0
+        read_buffer = []
+        for line in f:
+            read_buffer.append(line)
+            if len(read_buffer) == 4:  # Once we have a full read
+                if read_idx in sampled_indices:
+                    sampled_reads.extend(read_buffer)  # Add the read if it matches the sampled index
+                read_buffer = []  # Clear the buffer for the next read
+                read_idx += 1  # Move to the next read
+
+                # Stop if we've collected enough reads
+                if len(sampled_reads) >= sample_size * 4:
+                    break
+
+    return sampled_reads, sampled_indices
```
you should first merge my PR #28 to get a much faster version of this function + documentation!

sjanssen2 commented on vxdetector/VXdetector.py:
```diff
@@ -4,246 +4,207 @@
+        f.writelines(sampled_reads)
 
 
 def do_statistic(result):
```
it's hard for me to see what these changes are good for? Is this "just" refactoring the code OR also changing the functionality. You might want to make sure the github actions pass as they should include some unit tests

jeremymeiss committed:
Added Docstring to the missing Methods. Added
the sample_fastq method from the memory_efficient_
sampling branch. Sampled indices is no longer used
in sample_fastq, yet needed for the workflow to
funtion.

jeremymeiss committed:
Formatting

jeremymeiss committed:
more formatting

jeremymeiss committed:
formatted for flake8

jeremymeiss committed:
Delete .DS_Store

jeremymeiss committed:
Update .gitignore

ignore DS_Store

jeremymeiss committed:
Delete vxdetector/testlog

jeremymeiss committed:
Update environment.yml

jeremymeiss committed:
Update environment.yml

jeremymeiss committed:
Update environment.yml

Anna-Rehm commented:
It seems like pynose can't be installed using conda. You will most likely have to change parts of the test.yml https://github.com/jlab/algorithm_vxdetector/blob/2b2a01aab7acbfdc61fabcec1205a10475d2905c/.github/workflows/github_tests.yml#L18 and add the installation of the package via pip + remove the respective package from the environment.yml at the same time.

jeremymeiss committed:
Update environment.yml

jeremymeiss committed:
Update environment.yml

jeremymeiss committed:
Update VXdetector.py

jeremymeiss committed:
Update test_VXdetector.py

jeremymeiss committed:
Update test_VXdetector.py

jeremymeiss committed:
Merge branch 'master' of github.com:jlab/algorithm_vxdetector into Merged_Tool

jeremymeiss committed:
Merge branch 'Merged_Tool' of github.com:jlab/algorithm_vxdetector into Merged_Tool

jeremymeiss committed:
cleanuo

jeremymeiss committed:
TestChanges

jeremymeiss committed:
added biom to env

jeremymeiss committed:
-format

jeremymeiss committed:
Changed test to include new params

jeremymeiss committed:
Formatiing for Lint

jeremymeiss committed:
Changed the interact_bowtie2 Script

jeremymeiss committed:
formatting of interact_bowtie2

jeremymeiss committed:
fixes for testing

jeremymeiss committed:
reduced sample size for the test

jeremymeiss committed:
changed a parameter

jeremymeiss committed:
Changes to test_interact_bowtie

jeremymeiss committed:
undo changes

sjanssen2 commented:
Hi @Landbarsch ich sehe, dass Du fleißig versuchst die Tests zu korrigieren. Es scheint gar nicht so einfach zu sein. Vielleicht liegt es daran, dass Du zu viele Änderungen in einem versuchst zu beherrschen.

Ich sehe z.B. Änderungen in den Funktionen `workflow` und `do_statistic`. Aber haben die direkt etwas miteinander zu tun? Vielleicht machst Du erstmal noch einen neuen Branch auf, in dem Du **nur** die Änderungen an `do_statistic` versuchst einzupflegen. Wenn das klappt, könnte man diese Änderungen in den Branch hier hinein mergen und hätte einen übersichtlicheren Pull Request.
